### PR TITLE
Feature/it

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

**Comment by henryju** [comment]
LGTM

File: src/main/java/org/sonarlint/cli/report/IssuesReport.java [hunk 1]
```diff
@@ -85,11 +86,21 @@ public void setDate(Date date) {
...
     return new ArrayList<>(resourceReportsByFilePath.keySet());
   }
+  
+  public String getRuleName(String ruleKey) {
+    return ruleNameByKey.get(ruleKey);
+  }
+  
+  public String issueId(IssueListener.Issue issue) {
+    String line = issue.getStartLine() != null ? issue.getStartLine().toString() : "-";
+    return issue.getRuleKey() + "R" + issue.getFilePath().toString() + "L" + line;
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/cli/report/IssuesReport.java]: BLOCKER NullPointerException might be thrown as 'issue' is nullable here [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)

File: src/main/java/org/sonarlint/cli/report/ResourceReport.java [hunk 2]
```diff
@@ -61,7 +62,7 @@ public String getName() {
   }
 
   public String getType() {
-    if (filePath == null) {
+    if (filePath == null || filePath == Paths.get("")) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/cli/report/ResourceReport.java]: MAJOR Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**